### PR TITLE
add FiberMap & FiberSet modules

### PR DESCRIPTION
**Pull request by tim-smart**

### CODE REVIEW

File: packages/effect/src/FiberMap.ts [hunk 1]
```diff
@@ -0,0 +1,235 @@
...
+  },
+  [Inspectable.NodeInspectSymbol](this: FiberMap<unknown>) {
+    return this.toJSON()
+  },
+  pipe() {
+    return pipeArguments(this, arguments)
+  }
+}
+
+const unsakeMake = <K, E = unknown, A = unknown>(): FiberMap<K, E, A> => {
```
Comment (tim-smart) [inline packages/effect/src/FiberMap.ts:67]: unsafeMake*

File: packages/effect/src/FiberSet.ts [hunk 2]
```diff
@@ -0,0 +1,166 @@
...
+  },
+  [Inspectable.NodeInspectSymbol](this: FiberSet) {
+    return this.toJSON()
+  },
+  pipe() {
+    return pipeArguments(this, arguments)
+  }
+}
+
+const unsakeMake = <E = unknown, A = unknown>(): FiberSet<E, A> => {
```
Comment (tim-smart) [inline packages/effect/src/FiberSet.ts:63]: `unsafeMake`

File: packages/effect/src/FiberMap.ts (hunk 1 above)
Comment (fubhy) [reply packages/effect/src/FiberMap.ts:67]: I kinda like `unsakeMake`. Or what about `unlovinglyMake`

**Comment by mikearnaldi** [comment]
Rationale for the modules?

**Comment by tim-smart** [comment]
Discord discussion: https://discord.com/channels/795981131316985866/1194743674567934103/1194775331836547122

We thought managing a set of fibers has come up enough that having an official primitive could be beneficial.

**Comment by mikearnaldi** [comment]
maybe a bit of docs in jsdoc comments may help understanding usage

**Comment by tim-smart** [comment]
> maybe a bit of docs in jsdoc comments may help understanding usage

Updated the jsdoc

**Pull request merged**